Makefile

```diff
@@ -84,20 +132,20 @@ MY_UID:=$(shell id -u)
 MY_GID:=$(shell id -g)
 
 # Build a docker image used for building our go code into a binary.
-.PHONY: golang-build-image
-golang-build-image:
+.PHONY: calico-build/golang
+calico-build/golang:
```

Comment:
Like the streamlining of naming.  Couple of thoughts on the scheme though:
- our dockerhub prefix is `calico/` so we wouldn't be able to push these to dockerhub for caching purposes without retagging them (unless we grab calico-build too).
- should we put felix in there somewhere to distinguish from other projects and avoid name clashes?